Improve gated models doc

Wauplin opened the pull request:
Related to https://github.com/huggingface/hub-docs/issues/642. 

[**Updated Gated Models docs here.**](https://moon-ci-docs.huggingface.co/docs/hub/pr_1128/en/models-gated)

I spent some time revisiting the docs for gated models. Once reviewed I'll port it to the Gated Datasets page as well. I was even wondering if we wanted to have two different pages given that the content is 95% the same (`sed 's/model/dataset/g'` :smile:).

I followed the plan from https://github.com/huggingface/hub-docs/issues/642#issuecomment-1822807736:
> (nit) for me would be more logical in that order:
> - Intro - why gated models
> - How to manage gated models
> - How to use gated models (just clicking things through form, logging in, etc)

**TODO:**
- [x] gated models
- [ ] gated datasets

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/hub/pr_1128). All of your documentation changes will be reflected on that endpoint.

osanseviero commented on docs/hub/models-gated.md:
```diff
@@ -1,31 +1,74 @@
-To give model creators more control over how their models are used, the Hub allows users to enable **User Access requests** through a model's **Settings** tab.
-Enabling this setting requires users to agree to share their contact information and accept the model owners' terms and conditions in order to access the model.
-The contact information is stored in a database, and model owners are able to download a copy of the user access report.
+To give more control over how models are used, the Hub allows model authors to enable **Access requests** for their models. When enabled, users have to agree to share their contact information (username and email address) with the model authors to access the model files. A model with Access requests enabled is called a **Gated model**. Access requests are always granted to individual users and no ...
```
```suggestion
To give more control over how models are used, the Hub allows model authors to enable **access requests** for their models. When enabled, users have to agree to share their contact information (username and email address) with the model authors to access the model files. Model authors can configure this request with additional fields. A model with Access requests enabled is called a **Gated model**. Access requests are always granted to individual users and not to entire organizations. A common use case of gated models is to provide access to early research models before the wider release.
```

osanseviero commented on docs/hub/models-gated.md:
````diff
@@ -1,31 +1,74 @@
-from huggingface_hub import notebook_login
-notebook_login()
-```
+![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/models-gated-disabled.png)
````
Please also add darkmode images

Wauplin replied:
Thanks for forbidding me to be lazy :smile: Done in https://github.com/huggingface/hub-docs/pull/1128/commits/91c17ee8a0f8d23e8bd48c21b8da98a55aba4ee7.

osanseviero commented on docs/hub/models-gated.md:
```diff
@@ -1,31 +1,74 @@
+### Configure access control
 
-If you're using a Jupyter or Colaboratory notebook, log in with the following Python command
+To enable Access requests, go to the model settings page. By default the model is not gated. Click on "Enable Access request" on the top-right corner.
```
let's do lowecase (access request) consistently I think

osanseviero commented on docs/hub/models-gated.md:
```diff
@@ -1,31 +1,74 @@
+
+![](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/models-gated-user-side.png)
+
+If you want to get more information, you can configure additional fields for the user to fill. These information will be accessible from the settings tab to take the decision on whether or not to grant access. To do so, you should add a `extra_gated_fields` property to your [modelcard metadata](./model-cards#model-card-metadata) containing a list of key/value pairs. The *key* is the name of the f ...
```
```suggestion
If you want users to provide more information, you can configure additional fields for the user to fill. This information will be accessible from the settings tab to make the decision on whether or not to grant access. To do so, you should add an `extra_gated_fields` property to your [model card metadata](./model-cards#model-card-metadata) containing a list of key/value pairs. The *key* is the name of the field and *value* its type. A field can be either `text` (free text area) or `checkbox`. Finally, you can also personalize the message displayed to the user with the `extra_gated_prompt` extra field.
```

osanseviero commented on docs/hub/models-gated.md:
```diff
@@ -35,60 +78,48 @@ extra_gated_fields:
 
-By default, notifications for new pending access requests are sent once a day via email.
-When the repo lives in an organization, those emails are sent to the first 5 admins of the organization.
+Once the access request is sent, there are two possibilities. If the approval mechanism is automatic and you immediately get access to the model files. Otherwise, the requests have to be approved manually by the authors which can take more time. 
```
```suggestion
Once the access request is sent, there are two possibilities. If the approval mechanism is automatic, you immediately get access to the model files. Otherwise, the requests have to be approved manually by the authors, which can take more time. 
```

osanseviero reviewed (commented):
I love these! Thanks a lot! :fire:

Wauplin commented:
Thanks for the review @osanseviero! I've addressed all of your comments :)

osanseviero commented on docs/hub/models-gated.md:
```diff
@@ -1,94 +1,144 @@
+Those endpoints are not officially supported in `huggingface_hub` or `huggingface.js` yet but [this code snippet](https://github.com/huggingface/huggingface_hub/issues/1535#issuecomment-1614693412) (in Python) might help you getting started.
 
-### Notifications settings
+### Download access report
```
This change means that old urls pointing here won't work. I think you can add something like https://github.com/huggingface/transformers/blob/eb849f6604c7dcc0e96d68f4851e52e253b9f0e5/docs/source/en/model_summary.md?plain=1#L79 to make sure old URLs still work

Wauplin replied:
@osanseviero I added backward compatible anchors in in https://github.com/huggingface/hub-docs/pull/1128/commits/79e4e99b2a581728e117fd06010a43883d64fe98. I used `<a id="...."></a>` as done [here](https://github.com/huggingface/hub-docs/blob/81c6ad79680639f5ed8ee7f1bf260993821c4c61/docs/hub/spaces-sdks-docker.md?plain=1#L25C3-L25C3).

I think using `### Encoder[[audio-encoder]]` actually changes the anchor of the title. What we want here is to be backward compatible but not necessarily change the new anchor.

(to be confirmed...)

Wauplin replied:
It works: https://moon-ci-docs.huggingface.co/docs/hub/pr_1128/en/models-gated#modifying-the-prompt

pcuenca replied:
Makes sense to me. This works, as you said: https://huggingface.co/docs/hub/spaces-sdks-docker#secret-management (but the heading is scrolled out, maybe we should place it _before_ the heading :) )

Wauplin replied:
Arf indeed :grimacing: . Updated them in https://github.com/huggingface/hub-docs/pull/1128/commits/5f3b97fa98710852d913b2076b48ba04b35ebf06

osanseviero reviewed (approved):
Awesome, thank you! :fire:

Wauplin commented:
Thanks for the review and feedback @osanseviero! I'll let it open for a few days and will merge if I don't have new comments to address :)

pcuenca commented on docs/hub/models-gated.md:
````diff
@@ -1,94 +1,149 @@
----
-```
+If you want to manually approve which users can access your model, you must set it to **manual approval**. When this is the case, you will notice more options:
+- **Add access** allows you to search for a user and grant them access even if they did not requested it.
````
```suggestion
- **Add access** allows you to search for a user and grant them access even if they did not request it.
```

pcuenca commented on docs/hub/models-gated.md:
````diff
@@ -1,94 +1,149 @@
-```
+If you want to manually approve which users can access your model, you must set it to **manual approval**. When this is the case, you will notice more options:
+- **Add access** allows you to search for a user and grant them access even if they did not requested it.
+- **Notification frequency** let you configure when to get notified if new users request access. It can be set to once a day or real-time. By default, an email is sent to your primary email address. You can set a different email address in the **Notifications email** field. For models hosted under an organization, emails are sent to the first 5 admins of the organization.
````
```suggestion
- **Notification frequency** lets you configure when to get notified if new users request access. It can be set to once a day or real-time. By default, an email is sent to your primary email address. You can set a different email address in the **Notifications email** field. For models hosted under an organization, emails are sent to the first 5 admins of the organization.
```

pcuenca commented on docs/hub/models-gated.md:
```diff
@@ -1,94 +1,149 @@
-Only users whose access requests have been approved will be able to access the model's content.
+### From the UI
+
+You can review who have access to your gated model from its setting page by clicking on the **Review access requests** button. This will open a modal with 3 lists of users:
```
```suggestion
You can review who has access to your gated model from its settings page by clicking on the **Review access requests** button. This will open a modal with 3 lists of users:
```

pcuenca commented on docs/hub/models-gated.md:
```diff
@@ -1,94 +1,149 @@
+
+You can review who have access to your gated model from its setting page by clicking on the **Review access requests** button. This will open a modal with 3 lists of users:
+- **pending**: the list of users that are waiting for an approval to access your model. This list is empty unless you've selected **manual approval**. You can either "Accept" or "Reject" the demand. If the demand is rejected, the user cannot access your model and cannot request access again.
+- **accepted**: the complete list of users that have access to your model. You can chose to "Reject" the access at any time for any user, no matter if the approval mode is manual or automatic. You can also "Cancel" the approval which will move the user to the *pending* list.
```
```suggestion
- **accepted**: the complete list of users that have access to your model. You can choose to "Reject" the access at any time for any user, no matter if the approval mode is manual or automatic. You can also "Cancel" the approval, which will move the user to the *pending* list.
```

pcuenca commented on docs/hub/models-gated.md:
```diff
@@ -1,94 +1,149 @@
+
+### Request access
+
+As a user, if you want to use a gated model, you will need to request access to it. This mean that only registered users can access gated models.
```
```suggestion
As a user, if you want to use a gated model, you will need to request access to it. This means that only registered users can access gated models.
```

pcuenca commented on docs/hub/models-gated.md:
```diff
@@ -1,94 +1,149 @@
+    <img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/models-gated-user-side-dark.png"/>
+</div>
+
+By clicking on `Agree`, you agree to share your username and email address to the model authors. In some cases, additional fields might be requested. Try to fill the form as accurately as possible to help the model authors decide on whether they grant you the access or not.
```
```suggestion
By clicking on `Agree`, you agree to share your username and email address with the model authors. In some cases, additional fields might be requested. Try to fill the form as accurately as possible to help the model authors decide whether they grant you access or not.
```

pcuenca commented on docs/hub/models-gated.md:
```diff
@@ -1,94 +1,149 @@
+
+### Download files
+
+To download files from a gated model you'll need to be authenticated. In the browser, this is automatic as long as you are logged in with your account. If you are using a script, you will need to provide a [user token](./security-tokens). In the Python ecosystem (`transformers`, `diffusers`, `datasets`, etc.), you can login your machine using the [`huggingface_hub`](https://huggingface.co/docs/hu ...
```
```suggestion
To download files from a gated model you'll need to be authenticated. In the browser, this is automatic as long as you are logged in with your account. If you are using a script, you will need to provide a [user token](./security-tokens). In the Hugging Face Python ecosystem (`transformers`, `diffusers`, `datasets`, etc.), you can login your machine using the [`huggingface_hub`](https://huggingface.co/docs/huggingface_hub/index) library and running in your terminal:
```

pcuenca commented on docs/hub/models-gated.md:
````diff
@@ -1,94 +1,149 @@
+>>> login()
+```
+
+Finally, you can also provide the `token` parameter to any method of the Hugging Face ecosystem (`from_pretrained`, `hf_hub_download`, `load_dataset`, etc.) directly from your scripts.
````
```suggestion
You can also provide the `token` parameter to most loading methods in the libraries (`from_pretrained`, `hf_hub_download`, `load_dataset`, etc.), directly from your scripts.
```

pcuenca reviewed (approved):
Awesome, many thanks! Just suggested some typo fixes.

SBrandeis commented on docs/hub/models-gated.md:
```diff
@@ -1,94 +1,149 @@
+
+### Request access
+
+As a user, if you want to use a gated model, you will need to request access to it. This means that only registered users can access gated models.
```
Maybe more explicit this way (not sure)

```suggestion
As a user, if you want to use a gated model, you will need to request access to it. This means that you must be logged in to a Hugging Face user account to access gated models.
```

Wauplin commented on docs/hub/models-gated.md:
```diff
@@ -1,94 +1,148 @@
+<Tip warning={true}>
+
+The model authors have complete control over model access. In particular, they can decide at any time to block your access to the model without prior notice, regardless of approval mechanism or if your request has already been approved.
+
```
```suggestion

</Tip>

```

Wauplin commented:
Thanks everyone for the feedback on this PR :heart: I've addressed all the comments and will now merge it!
And I'll open a second PR to transpose this to the `datasets-gated.md` page as well.
